kasa/sslaestransport.py

```diff
@@ -0,0 +1,443 @@
+"""Implementation of the TP-Link AES transport.
+
+Based on the work of https://github.com/petretiandrea/plugp100
+under compatible GNU GPL3 license.
+"""
+
+from __future__ import annotations
+
+import base64
+import hashlib
+import logging
+import secrets
+import ssl
+import time
+from enum import Enum, auto
+from typing import TYPE_CHECKING, Any, Dict, cast
+
+from cryptography.hazmat.primitives import padding
+from cryptography.hazmat.primitives.ciphers import Cipher, algorithms, modes
+from urllib3.util import create_urllib3_context
+from yarl import URL
+
+from .aestransport import AesEncyptionSession
+from .credentials import Credentials
+from .deviceconfig import DeviceConfig
+from .exceptions import (
+    SMART_AUTHENTICATION_ERRORS,
+    SMART_RETRYABLE_ERRORS,
+    AuthenticationError,
+    DeviceError,
+    KasaException,
+    SmartErrorCode,
+    _RetryableError,
+)
+from .httpclient import HttpClient
+from .json import dumps as json_dumps
+from .json import loads as json_loads
+from .protocol import BaseTransport
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ONE_DAY_SECONDS = 86400
+SESSION_EXPIRE_BUFFER_SECONDS = 60 * 20
+
+
+def _sha1(payload: bytes) -> str:
+    sha1_algo = hashlib.sha1()  # noqa: S324
+    sha1_algo.update(payload)
+    return sha1_algo.hexdigest()
+
+
+def _sha256(payload: bytes) -> bytes:
+    return hashlib.sha256(payload).digest()  # noqa: S324
+
+
+def _md5_hash(payload: bytes) -> str:
+    return hashlib.md5(payload).hexdigest().upper()  # noqa: S324
+
+
+def _sha256_hash(payload: bytes) -> str:
+    return hashlib.sha256(payload).hexdigest().upper()  # noqa: S324
```

Comment:
Plenty of shared utils here. Are the differences so numerous, that it's not worth try to adapt the aestransport to add support for ssl? Not that important right now, but worth checking out at some point.

Reply:
Agreed